zypp/sat/detail/PoolImpl.cc

```diff
@@ -505,8 +505,23 @@ namespace zypp
 	  // Add removed locales+fallback except they are still in current
 	  for ( Locale lang: localesTracker.removed() )
 	  {
-	    for ( ; lang && ! localeIds.current().count( IdString(lang) ); lang = lang.fallback() )
-	    { localeIds.removed().insert( IdString(lang) ); }
+	    if ( lang && ! localeIds.current().count( IdString(lang) ) ) {
+
+	      //remove the requested one
+	      localeIds.removed().insert( IdString(lang) );
+
+	      //remove fallbacks
+	      const auto &currLangs = localeIds.current();
+	      auto checkIsFallback = [&lang]( const IdString &currLoc_r ){
+		return Locale(currLoc_r).hasFallback(lang);
+	      };
+	      for ( lang = lang.fallback(); lang && ! localeIds.current().count( IdString(lang) ); lang = lang.fallback() ) {
+		//remove the language only if its not a fallback for any other
+		if ( std::none_of( currLangs.begin(), currLangs.end(), checkIsFallback ) )
```

Comment:
No longer sure if this is the right approach / whether `checkIsFallback` can actually happen.

The 1st of the 3 for-loops (L490) adds all already supported locales (in `localesTracker.current` but nor newly added) _and their fallbacks_ to `localeIds.current()`.

The 2nd loop (L499) adds all newly added locales including their fallbacks to `localeIds.added()`, unless a locale is already in `localeIds.current()` (in this case it's fallbacks are also already in current()).

IMO the original 3rd-loop is almost right. All removed locales and their fallbacks are remembered in `localeIds.removed()` unless a fallback is already in `localeIds.current()` .  It just misses an additional  check for ` && ! localeIds.added().count( IdString(lang) )`. The fallback must not be removed it was newly added (removed `de_DE` but added `de`).

In contrast to `localesTracker`, `localeIds.current()` contains all fallbacks. So if a locale is not in current, it can't be fallback of a locale in current.